src/metaswitch/clearwater/cluster_manager/etcd_synchronizer.py

```diff
@@ -155,195 +122,19 @@ def mark_node_failed(self):
-                        break
-                    except urllib3.exceptions.TimeoutError:
-                        # Timeouts after 5 seconds are expected, so ignore them
-                        # - unless we're terminating, we'll stay in the while
-                        # loop and try again
-                        pass
-                    except etcd.EtcdException as e:
-                        # We have seen timeouts getting raised as EtcdExceptions
-                        # so catch these here too and treat them as timeouts if
-                        # they indicate that the read timed out.
-                        if "Read timed out" in e.message:
-                            pass
-                        else:
-                            raise
-                    except ValueError:
-                        # The index isn't valid to watch on, probably because
-                        # there has been a snapshot between the get and the
-                        # watch. Just start the read again.
-                        _log.info("etcd index {} is invalid, retrying".format(
-                            result.modifiedIndex+1))
-                        self.read_from_etcd()
-
-                # Return if we're terminating.
-                if self._terminate_flag:
-                    return
-                else:
-                    cluster_view = self.parse_cluster_view(result.value)
-
-            # Save off the index of the result we're using for when we write
-            # back to etcd later.
-            self._index = result.modifiedIndex
-            self._last_cluster_view = cluster_view.copy()
-
-        except etcd.EtcdKeyError:
-            _log.info("Key {} doesn't exist in etcd yet".format(self._key))
-            # If the key doesn't exist in etcd then there is currently no
-            # cluster.
-            cluster_view = {}
-            self._index = None
-            self._last_cluster_view = None
-            if not self._plugin.should_be_in_cluster():
-                # We're watching a key managed by another plugin, and it
-                # doesn't exist yet - sleep to give the other plugin a chance to
-                # create it
-                sleep(self.PAUSE_BEFORE_RETRY)
-        except Exception as e:
-            # Catch-all error handler (for invalid requests, timeouts, etc -
-            # start over.
-            _log.error("{} caught {!r} when trying to read with index {}"
-                       " - pause before retry".
-                       format(self._ip, e, self._index))
-            # Sleep briefly to avoid hammering a failed server
-            sleep(self.PAUSE_BEFORE_RETRY)
-            # The main loop (which reads from etcd in a loop) should call this
-            # function again after we return, causing the read to be retried.
+        etcd_result, idx = self.read_from_etcd(wait=False)
+        print etcd_result
+        assert(etcd_result is not None)
```

Comment:
If we're going to assert here then (which I think is when someone tries to mark a non-existent node as failed), then there should be a useful error message